### ISSUE TITLE
Crash with x11 backend when trying to swap containers

### ISSUE THREAD

**Issue by imirkin** [swaywm/sway#5962]
See backtrace below. I'm running the latest HEAD sway/wlroots. I'm using the x11 backend (running on plain Xorg), and haven't installed sway. I'm running sway like this:

```
PATH=build/swaybar:build/swaymsg:build/swaynag:$PATH ./build/sway/sway -c build/config -d
```

I started 2x weston-terminal's, and then dragged one onto the other.

```
00:00:22.636 [sway/commands/swap.c:108] Swapping containers 5 and 6

Thread 1 "sway" received signal SIGSEGV, Segmentation fault.
0x000055555558db96 in seat_get_focus (seat=0x0) at ../sway/input/seat.c:1377
1377            if (!seat->has_focus) {
(gdb) bt
#0  0x000055555558db96 in seat_get_focus (seat=0x0)
    at ../sway/input/seat.c:1377
#1  0x000055555558dcfb in seat_get_focused_container (seat=0x0)
    at ../sway/input/seat.c:1417
#2  0x00005555555aad84 in container_swap
    (con1=0x555555ce4da0, con2=0x555555d49070) at ../sway/commands/swap.c:138
#3  0x0000555555591b80 in finalize_move (seat=0x555555c3bd30)
    at ../sway/input/seatop_move_tiling.c:255
#4  0x0000555555591d97 in handle_button
    (seat=0x555555c3bd30, time_msec=33608206, device=0x555555ca4850, button=272, state=WLR_BUTTON_RELEASED) at ../sway/input/seatop_move_tiling.c:304
#5  0x000055555558e0d2 in seatop_button
    (seat=0x555555c3bd30, time_msec=33608206, device=0x555555ca4850, button=272, state=WLR_BUTTON_RELEASED) at ../sway/input/seat.c:1506
#6  0x0000555555583a7c in dispatch_cursor_button
    (cursor=0x555555c3c2d0, device=0x555555ca4850, time_msec=33608206, button=272, state=WLR_BUTTON_RELEASED) at ../sway/input/cursor.c:414
#7  0x0000555555583b5c in handle_pointer_button
    (listener=0x555555c3c430, data=0x7fffffffd5d0)
    at ../sway/input/cursor.c:432
#8  0x00007ffff7f7dff0 in wlr_signal_emit_safe
    (signal=0x555555c3c5d8, data=0x7fffffffd5d0)
    at ../subprojects/wlroots/util/signal.c:29
#9  0x00007ffff7f55064 in handle_pointer_button
    (listener=0x555555c8a430, data=0x7fffffffd5d0)
    at ../subprojects/wlroots/types/wlr_cursor.c:437
#10 0x00007ffff7f7dff0 in wlr_signal_emit_safe
    (signal=0x555555ca47b8, data=0x7fffffffd5d0)
    at ../subprojects/wlroots/util/signal.c:29
#11 0x00007ffff7f38857 in send_button_event
    (output=0x555555ca44d0, key=272, st=WLR_BUTTON_RELEASED, time=33608206)
    at ../subprojects/wlroots/backend/x11/input_device.c:41
#12 0x00007ffff7f38edb in handle_x11_xinput_event
    (x11=0x5555555fbb60, event=0x555555c94a60)
    at ../subprojects/wlroots/backend/x11/input_device.c:186
#13 0x00007ffff7f36beb in handle_x11_event
    (x11=0x5555555fbb60, event=0x555555c94a60)
    at ../subprojects/wlroots/backend/x11/backend.c:108
#14 0x00007ffff7f36cc2 in x11_event (fd=4, mask=1, data=0x5555555fbb60)
    at ../subprojects/wlroots/backend/x11/backend.c:137
#15 0x00007ffff78bc412 in wl_event_loop_dispatch ()
    at /usr/lib64/libwayland-server.so.0
#16 0x00007ffff78ba9f5 in wl_display_run ()
    at /usr/lib64/libwayland-server.so.0
#17 0x0000555555574add in server_run (server=0x5555555eb320 <server>)
    at ../sway/server.c:247
#18 0x0000555555573fbb in main (argc=4, argv=0x7fffffffdaa8)
    at ../sway/main.c:431
```

**Comment by imirkin** [comment]
This seems similar to #5939

**Comment by imirkin** [comment]
Yeah. This issue is also fixed by PR #5950.
